test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc

```diff
@@ -86,6 +87,127 @@ TEST(QuicPlatformTest, QuicPtrUtil) {
   EXPECT_EQ("aaa", *p);
 }
 
+namespace {
+class QuicLogThresholdSaver {
+public:
+  QuicLogThresholdSaver()
+      : level_(QUIC_LOGGER().level()), verbosity_threshold_(quic::GetVerbosityLogThreshold()) {}
+
+  ~QuicLogThresholdSaver() {
+    quic::SetVerbosityLogThreshold(verbosity_threshold_);
+    QUIC_LOGGER().set_level(level_);
+  }
+
+private:
+  const quic::QuicLogLevel level_;
+  const int verbosity_threshold_;
+};
+} // namespace
+
+TEST(QuicPlatformTest, QuicLog) {
+  QuicLogThresholdSaver saver;
+
+  // By default, tests emit logs at level ERROR or higher.
+  ASSERT_EQ(quic::ERROR, QUIC_LOGGER().level());
+
+  int i = 0;
+
+  QUIC_LOG(INFO) << (i = 10);
+  QUIC_LOG_IF(INFO, false) << i++;
+  QUIC_LOG_IF(INFO, true) << i++;
+  EXPECT_EQ(0, i);
+
+  QUIC_LOG(ERROR) << (i = 11);
+  EXPECT_EQ(11, i);
+
+  QUIC_LOG_IF(ERROR, false) << i++;
+  EXPECT_EQ(11, i);
+
+  QUIC_LOG_IF(ERROR, true) << i++;
```

Comment:
would it be possible to have the test verify, for at least one of the log statements that actually executes, that the expected message actually makes it to spdlog? Might not be too bad if you mock spdlog::sinks::sink or base_sink.

If this ends up being more complicated, then maybe not worth the effort.

Reply:
I plan to work on it in a follow up PR, in the context of quic_mock_log_impl.h.

Reply:
SG. Could you add a TODO comment for that?

Reply:
Done, thanks!